### PR TITLE
Add additional mountPaths and logging file paths for config reload container

### LINKED ISSUE #1139: Add capability to include custom volume mounts on config reload container
### Is your feature request related to a problem? Please describe

The helm chart currently allows one to a) add additional containers and b) additional volume mounts with .Values.controller.sidecars.additionalSidecarContainers and .Values.persistence.volumes respectively.  This allows the ability to create a Grafana Alloy sidecar container, configure the main `jenkins` container to log to a file in a mounted folder, and share this folder with Grafana Alloy for scraping (and filtering, etc). 

At this moment, the config reload container does not have the ability to add additional volume mounts.

### Describe the solution you'd like

Update the `jenkins.configReloadContainer` helper function to add the ability to add custom volume mounts into the container.

Additionally, provide some defaults for setting kiwigrid/k8s-sidecar `LOG_CONFIG` environment variable (the file path to this the log file) and the contents of the log config file itself (to override the default - https://github.com/kiwigrid/k8s-sidecar/blob/master/src/logger.py#L69)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jesrig**
### What does this PR do?

- Fixes #1139

If you modified files in the `./charts/jenkins/` directory, please also include the following:

```[tasklist]
### Submitter checklist
- [x] I bumped the "version" key in `./charts/jenkins/Chart.yaml`.
- [x] I added a new changelog entry to `./charts/jenkins/CHANGELOG.md`.
- [x] I followed the [technical requirements](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements).
- [x] I ran `.github/helm-docs.sh` from the project root.
```

### CODE REVIEW

**Comment by ml-software-services** [comment]
@kvanzuijlen I'm a first time contributor to this helm chart repository. Can you help me understand what I need to fix here? I manually deleted the untitest snapshot directory and reran the `ct` command. This updated the snapshot files, which I then committed to the repo. The linting errors shown above no longer are recreateable locally.

Please advise.

**Comment by jesrig** [comment]
@kvanzuijlen 
I am a first time contributor to this helm chart, can you help me understand what I need to do to fix this PR?
When ran into this linting error locally, I manually deleted the `__snapshot__` folder, ran `ct lint --config ct.yaml` and it passed. I then commited the changes to `__snapshot__` as part of my PR.

Please advise.

File: charts/jenkins/values.yaml [hunk 1]
```diff
@@ -567,6 +567,24 @@ controller:
...
+      additionalVolumeMounts: []
+        #   - name: auto-reload-config
+        #     mountPath: /var/config/logger
+        #   - name: auto-reload-logs
+        #     mountPath: /var/log/auto_reload
+      # -- Config auto-reload logging settings
+      logging:
+        configuration:
+          # -- Enables custom log config utilizing using the settings below.
+          # See default settings https://github.com/kiwigrid/k8s-sidecar/blob/master/src/logger.py
```
Comment (kvanzuijlen) [inline charts/jenkins/values.yaml:580]: please move this line above the docs line

**Comment by kvanzuijlen** [comment]
@jesrig I'll try to take a look later today

File: charts/jenkins/templates/jenkins-controller-statefulset.yaml [hunk 2]
```diff
@@ -325,6 +325,13 @@ spec:
 {{- if .Values.persistence.volumes }}
 {{ tpl (toYaml .Values.persistence.volumes | indent 6) . }}
 {{- end }}
+      {{- if .Values.controller.sidecars.configAutoReload.logging.configuration.override }}
```
Comment (kvanzuijlen) [inline charts/jenkins/templates/jenkins-controller-statefulset.yaml:328]: Add a test case to test these values. You now added values to the snapshot without adding a test configuration for the configAutoReload configuration override.

**Comment by jesrig** [comment]
@kvanzuijlen All checks now passed, and I believe I addressed all of your comments. If you could re-review, that would be great.

**Review by timja — APPROVED** [review]

**Pull request merged**